src/v2/System/Router/RouterLink.tsx

```diff
@@ -1,93 +1,73 @@
-import { Link, LinkProps, LinkPropsSimple, RouterContext } from "found"
-import { omit, pick } from "lodash"
+import { Link, LinkProps, RouterContext } from "found"
 import React, { useContext } from "react"
-import { get } from "v2/Utils/get"
+import { BoxProps, boxMixin } from "@artsy/palette"
+import styled from "styled-components"
+import { compose, ResponsiveValue, system } from "styled-system"
+import { useMemo } from "react"
 
 /**
  * Wrapper component around found's <Link> component with a fallback to a normal
  * <a> tag if ouside of a routing context.
- *
- * NOTE: If using styled-components, <RouterLink> can be easily styled like so:
- *
- * const StyledLink = styled(RouterLink)`
- *   ...
- * `
  */
-export type RouterLinkProps = LinkProps &
-  React.HTMLAttributes<HTMLAnchorElement> & {
+export type RouterLinkProps = Omit<
+  React.AnchorHTMLAttributes<HTMLAnchorElement>,
+  "href"
+> &
+  Omit<LinkProps, "to"> &
+  BoxProps & {
+    /**
+     * Simplifies `LinkProps#to` to just be a string and handle nulls, which are common
+     */
+    to: string | null
+    textDecoration?: ResponsiveValue<string>
+    /** @deprecated Use `textDecoration` */
     noUnderline?: boolean
   }
 
 export const RouterLink: React.ForwardRefExoticComponent<RouterLinkProps> = React.forwardRef(
-  ({ to, children, ...props }, ref) => {
-    const styleProps = props.noUnderline ? { textDecoration: "none" } : {}
+  ({ to, noUnderline, ...rest }, forwardedRef) => {
     const context = useContext(RouterContext)
-    const routes = get(context, c => c?.router?.matcher?.routeConfig, [])
-    const isSupportedInRouter = !!get(context, c =>
-      c?.router?.matcher?.matchRoutes(routes, to)
+    const routes = context?.router?.matcher?.routeConfig ?? []
+    const matcher = context?.router?.matcher
+    const isSupportedInRouter = useMemo(
+      () => !!matcher?.matchRoutes(routes, to),
+      [matcher, routes, to]
     )
 
-    /**
-     * Only pass found-router specific props across, props that conform to the
-     * link API found here: https://github.com/4Catalyzer/found#links
-     */
-    const handlers = Object.keys(props).reduce((acc, prop) => {
-      if (prop.startsWith("on")) {
-        // @ts-expect-error STRICT_NULL_CHECK
-        acc.push(prop)
-      }
-      return acc
-    }, [])
+    // TODO: Bulk replace
+    const deprecated = noUnderline ? { textDecoration: "none" } : {}
 
     if (isSupportedInRouter) {
-      const allowedProps = pick(props, [
-        "aria-label",
-        "Component",
-        "activeClassName",
-        "className",
-        "data-test",
-        "exact",
-        "replace",
-        "role",
-        "style",
-        "tabIndex",
```

Comment:
Are we now getting way more warnings in the dom? Other react-ish props that are passed down used to trigger a waterfall of warnings in the console

Reply:
Nah, those stopped being a thing a while ago. (We can actually fix that correctly now in Styled Components 5, I believe)

Reply:
Ah, nice. What do we need to do in SC 5 (now that its already in?)

Reply:
lol I stand corrected, I was checking the browser console not the server console. Anyway we can fix this the right way sans safelist now: https://styled-components.com/docs/api#shouldforwardprop

> shouldForwardProp can take a second parameter that provides access to the default validator function. This function can be used as a fallback, and of course, it also works like a predicate, filtering based on known HTML attributes.